spyder/api/plugins.py

```diff
@@ -533,3 +557,874 @@ def set_ancestor(self, ancestor):
         Needed to update the ancestor/parent of child widgets when undocking.
         """
         pass
+
+
+##############################################################################
+#
+# New API, to migrate plugins one by one and make it easier to test changes
+#
+##############################################################################
+class SpyderPluginV2(QObject):
+    """
+    A Spyder plugin to enhance functionality without a dockable widget.
+
+    If you want to create a plugin that adds a new panel, use
+    SpyderDockableWidget.
+    """
+
+    # --- API: Mandatory attributes ------------------------------------------
+    # ------------------------------------------------------------------------
+    # Name of the plugin that will be used to refer plugins.
+    # This name must be unique and will only be loaded once.
+    NAME = None
+
+    # --- API: Optional attributes ------------------------------------------
+    # ------------------------------------------------------------------------
+    # List of required plugin dependencies names
+    # Example: ['plots', 'ipython_console', ...]
+    REQUIRES = None
+
+    # List of required plugin dependencies names
```

Comment:
required -> optional?

And why does it mean that a plugin has optional dependencies? Could give update the comment above with an example?

Reply:
Fixed